File: app/src/main/java/com/guideapp/ui/views/localdetail/LocalDetailContract.java
package com.guideapp.ui.views.localdetail;

import android.content.Context;
import android.support.v4.app.LoaderManager;

interface LocalDetailContract {
    interface View {
        void showFavoriteYes();

        void showFavoriteNo();

        void showSnackbarRemoveFavorite();

        void showSnackbarSaveFavorite();

        void shareText(String textToShare);

        Context getContext();

        void showTitle(String description);

        void showImage(String imagePath);

        void showCategory(String text);

        void showWebSiteAction();

        void showDirectionAction();

        void showCallAction();

        void showCall(String phone);

        void showDetail(String description);

        void showAddress(String address);

        void showMap(double latitude, double longitude, int idImageMarker);

        void dialPhoneNumber(String number);

        void openPage(String url);

        void openDirection(String description, String latLng);
    }

    interface Presenter {
        void loadLocal(LoaderManager loaderManager);

        void destroy(LoaderManager loaderManager);

        void saveOrRemoveFavorite();

        void shareLocal();

        void loadWebsite();

        void loadCall();

        void loadDirection();
    }
}

File: app/src/main/java/com/guideapp/ui/views/localdetail/LocalDetailContract.kt
package com.guideapp.ui.views.localdetail

import android.content.Context
import android.support.v4.app.LoaderManager

internal interface LocalDetailContract {
    interface View {
        fun showFavoriteYes()

        fun showFavoriteNo()

        fun showSnackbarRemoveFavorite()

        fun showSnackbarSaveFavorite()

        fun shareText(textToShare: String)

        fun getContext(): Context

        fun showTitle(description: String)

        fun showImage(imagePath: String)

        fun showCategory(text: String)

        fun showWebSiteAction()

        fun showDirectionAction()

        fun showCallAction()

        fun showCall(phone: String)

        fun showDetail(description: String)

        fun showAddress(address: String)

        fun showMap(latitude: Double, longitude: Double, idImageMarker: Int)

        fun dialPhoneNumber(number: String)

        fun openPage(url: String)

        fun openDirection(description: String, latLng: String)
    }

    interface Presenter {
        fun loadLocal(loaderManager: LoaderManager)

        fun destroy(loaderManager: LoaderManager)

        fun saveOrRemoveFavorite()

        fun shareLocal()

        fun loadWebsite()

        fun loadCall()

        fun loadDirection()
    }
}
